tcf_website/management/commands/index_elasticsearch.py

```diff
@@ -0,0 +1,113 @@
+"""
+Modules used
+"""
+import json
+import requests
+
+from django.core.management.base import BaseCommand, CommandError
+from tcf_website.models import Course, Instructor
+
+class Command(BaseCommand):
+    """Indexes the Elastic AppSearch instance w/ Course and Instructor.
+
+    Author: Bradley Knaysi
+
+    Created: April 19th, 2020
+
+    How To Use:
+
+        $ cd theCourseForum2/
+        $ docker-compose up
+
+        open a new terminal
+
+        $ cd theCourseForum2/
+        $ docker exec -it tcf_django bash
+        $ python3 manage.py index_elasticsearch
+
+    WARNING: This should only be done by an Executive Team member once a semester
+    after new course and instructor data are added to the tcf_db. Also the Elastic AppSearch
+    portal takes like 10 minutes to fully update so be patient there. You can run this as
+    many times as you want! It updates a document in Elastic if it always exists so won't
+    double add or anything weird like that!
+
+    """
+
+    help = 'Erases and re-indexes the Elastic hosted Elasticsearch cluster'
+
+    def handle(self, *args, **options):
+
+        courses_engine_endpoint = 'https://1761244fa3674318b11f8d5729ffd7a2.app-search.us-east4.gcp.elastic-cloud.com/api/as/v1/engines/uva-courses/documents'
+
+        # PRIVATE READ / WRITE KEY (need to find more secret place)
+        api_key = 'private-tw8dxda1rfxw28s7dxc6vihs'
+
+        https_headers = {
+            "Content-Type" : "application/json",
+            "Authorization" : "Bearer " + api_key
+        }
+
+        all_courses = Course.objects.all().order_by('pk')
```

Comment:
Should we re-index all courses or just courses last taught in the current semester?

Reply:
I say stick will all courses, because it will UPDATE if there are any changes and won't do anything if it's the same. This approach is more robust from future error and there will never be enough courses to make this an issue, even with 14000 courses it only takes ~120 seconds

Reply:
Sounds good to me!